src/workerd/jsg/ser.c++

```diff
@@ -41,22 +51,73 @@ v8::Maybe<uint32_t> Serializer::GetSharedArrayBufferId(
   return v8::Just(n);
 }
 
+void Serializer::scheduleDataCloneErrorForObject(v8::Isolate* isolate, v8::Local<v8::Object> obj) {
+  jsg::Lock& js = jsg::Lock::from(isolate);
+
+  try {
+    // The default error that V8 would generate is "#<TypeName> could not be cloned." -- for some
+    // reason, it surrounds the type name in "#<>", which seems bizarre? Let's generate a better
+    // error.
+    v8::Local<v8::String> message = js.str(kj::str(
+        "Could not serialize object of type \"", obj->GetConstructorName(), "\". This type does "
+        "not support serialization."));
+    isolate->ThrowException(makeDOMException(isolate, message, "DataCloneError"));
+  } catch (JsExceptionThrown&) {
+    // Apparently an exception was thrown during the construction of the DOMException. Most likely
+    // we were terminated. In any case, we'll let that exception stay scheduled and propagate back
+    // to V8.
+  }
```

Comment:
non-blocking for latre: Any chance of a `kj::Exception` being thrown? Should we handle it just in case?

Reply:
Cleaned this up: 0a3a2719